packages/react-scripts/config/paths.js

```diff
@@ -117,3 +118,40 @@ if (
   };
 }
 // @remove-on-eject-end
+
+module.exports.srcPaths = [module.exports.appSrc];
+
+// recursively find package dirs
+// -- omit things in node_modules, nested packages, and current app package
+const findPkgs = root =>
+  fs
+    .readdirSync(root)
+    .filter(f => ['node_modules', '.git'].indexOf(f) === -1)
+    .map(f => path.join(root, f))
+    .filter(f => fs.realpathSync(f) !== appDirectory)
+    .filter(f => fs.statSync(f).isDirectory())
+    .reduce(
+      (pkgs, dir) =>
+        fs.existsSync(path.join(dir, 'package.json'))
+          ? pkgs.concat([dir])
+          : pkgs.concat(findPkgs(dir)),
+      []
+    );
+
+if (checkForMonorepo) {
+  // if app is in a monorepo (lerna or yarn workspace), allow any module inside
+  // the monorepo to be linted, transpiled, and tested as if it came from app's
+  // src.
+  const monoPkgPath = findPkg.sync(resolveApp('..'));
```

Comment:
What if a project just happens to be located inside a monorepo but isn't actually a *part* of it (in the sense that it's not specified in `workspaces`)?